src/builder.rs

```diff
@@ -362,7 +362,20 @@ fn append_dir(dst: &mut Write, path: &Path, src_path: &Path, mode: HeaderMode) -
     append_fs(dst, path, &stat, &mut io::empty(), mode, None)
 }
 
-fn prepare_header(dst: &mut Write, header: &mut Header, path: &Path) -> io::Result<()> {
+fn prepare_header(size: u64, entry_type: u8) -> Header {
+    let mut header = Header::new_gnu();
+    header.as_gnu_mut().unwrap().name[..13].clone_from_slice(b"././@LongLink");
```

Comment:
While you're here (I realize this was just copied from before), mind cleaning this up a bit? Ideally the string and the 13 length it has would be deduplicated slightly to just store the string to a local variable and read its lenght

Reply:
done !